packages/venia-concept/src/components/OnlineIndicator/onlineIndicator.css

```diff
@@ -0,0 +1,67 @@
+.root {
+    align-items: center;
+    display: inline-flex;
+    font-size: 0.9em;
+    height: 0px;
+    justify-content: flex-start;
+    margin-bottom: 0.3em;
+    overflow: hidden;
+    padding: 0 2.4rem;
+    position: sticky;
+    top: 0;
+    width: 100%;
+}
+
+.offline {
+    composes: root;
+    background-color: rgb(var(--venia-warning-light));
+    animation-name: slidein;
+    animation-duration: 244ms;
+    animation-fill-mode: forwards;
+}
+
+.online {
+    composes: root;
+    background-color: rgb(var(--venia-teal-light));
+    animation-name: slideinandhide;
+    animation-duration: 3s;
+    animation-fill-mode: forwards;
+}
+
+.offline span {
+    color: rgb(var(--venia-warning-dark));
+}
+
+.online span {
+    color: rgb(var(--venia-teal));
+}
+
+.root p {
+    margin-left: 1em;
+    line-height: 1.3em;
+}
+
+@keyframes slidein {
+    from {
+        height: 0;
+    }
+    to {
+        height: 3rem;
+    }
+}
+
+@keyframes slideinandhide {
+    0% {
+        height: 3rem;
+        opacity: 0.5;
+    }
+    10% {
+        opacity: 1;
+    }
+    80% {
+        height: 3rem;
+    }
+    100% {
+        height: 0;
+    }
+}
```

Comment:
As Rowan had mentioned, animating `height` will cause render jank. I second his recommendationt to make it an overlay.

In addition, `height` is not a GPU-accelerated property, so we should avoid animating it. Try to restrict your animations to `transform` and `opacity`.